s2counter_core/src/main/scala/s2/counter/core/v2/RankingStorageGraph.scala

```diff
@@ -83,35 +84,39 @@ class RankingStorageGraph(config: Config) extends RankingStorage {
-          val prevRankingSeq = toWithScoreLs(edges)
-          val prevRankingMap: Map[String, Double] = prevRankingSeq.groupBy(_._1).map(_._2.sortBy(-_._2).head)
-          val currentRankingMap: Map[String, Double] = value.mapValues(_.score)
-          val mergedRankingSeq = (prevRankingMap ++ currentRankingMap).toSeq.sortBy(-_._2).take(k)
-          val mergedRankingMap = mergedRankingSeq.toMap
-
-          val bucketRankingSeq = mergedRankingSeq.groupBy { case (itemId, score) =>
-            // 0-index
-            GraphUtil.transformHash(MurmurHash3.stringHash(itemId)) % BUCKET_SHARD_COUNT
-          }.map { case (shardIdx, groupedRanking) =>
-            shardIdx -> groupedRanking.filter { case (itemId, _) => currentRankingMap.contains(itemId) }
-          }.toSeq
-
-          insertBulk(key, bucketRankingSeq).flatMap { _ =>
-            val duplicatedItems = prevRankingMap.filterKeys(s => currentRankingMap.contains(s))
-            val cutoffItems = prevRankingMap.filterKeys(s => !mergedRankingMap.contains(s))
-            val deleteItems = duplicatedItems ++ cutoffItems
-
-            val keyWithEdgesLs = prevRankingSeq.map(_._1).zip(edges)
-            val deleteEdges = keyWithEdgesLs.filter{ case (s, _) => deleteItems.contains(s) }.map(_._2)
-
-            deleteAll(deleteEdges)
+        if (checkAndPrepareDimensionBucket(key)) {
+          val future = getEdges(key, "raw").flatMap { edges =>
+            val prevRankingSeq = toWithScoreLs(edges)
+            val prevRankingMap: Map[String, Double] = prevRankingSeq.groupBy(_._1).map(_._2.sortBy(-_._2).head)
+            val currentRankingMap: Map[String, Double] = value.mapValues(_.score)
+            val mergedRankingSeq = (prevRankingMap ++ currentRankingMap).toSeq.sortBy(-_._2).take(k)
+            val mergedRankingMap = mergedRankingSeq.toMap
+
+            val bucketRankingSeq = mergedRankingSeq.groupBy { case (itemId, score) =>
+              // 0-index
+              GraphUtil.transformHash(MurmurHash3.stringHash(itemId)) % BUCKET_SHARD_COUNT
+            }.map { case (shardIdx, groupedRanking) =>
+              shardIdx -> groupedRanking.filter { case (itemId, _) => currentRankingMap.contains(itemId) }
+            }.toSeq
+
+            insertBulk(key, bucketRankingSeq).flatMap { _ =>
+              val duplicatedItems = prevRankingMap.filterKeys(s => currentRankingMap.contains(s))
+              val cutoffItems = prevRankingMap.filterKeys(s => !mergedRankingMap.contains(s))
+              val deleteItems = duplicatedItems ++ cutoffItems
+
+              val keyWithEdgesLs = prevRankingSeq.map(_._1).zip(edges)
+              val deleteEdges = keyWithEdgesLs.filter{ case (s, _) => deleteItems.contains(s) }.map(_._2)
+
+              deleteAll(deleteEdges)
+            }
           }
-        }
 
-        future
+          future
+        }
+        else {
+          // do nothing
+          Future.successful(false)
+        }
       }
```

Comment:
No, the original code does not check the return value of ```checkAndPrepareDimensionBucket```. So, I changed the code to check the return value.